error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/ImplicitBlockingFluxOperation.java

```diff
@@ -0,0 +1,192 @@
+        SuggestedFixes.qualifyStaticImport(
+            "java.util.stream.Collectors.toUnmodifiableList", imports, state);
+    String collector = toUnmodifiableList + "()";
+
+    return replaceMethodInvocationWithCollect(tree, collector, imports.build(), state);
+  }
+
+  // XXX: Assumes that the generated `collect(...)` expression will evaluate to
+  // `Mono<Collection<?>>`
+  // XXX: May return `emptyFix`.
+  private static SuggestedFix replaceMethodInvocationWithCollect(
+      MethodInvocationTree tree,
+      String collectArgument,
+      SuggestedFix additionalFixes,
+      VisitorState state) {
+    String collectMethodInvocation = String.format("collect(%s)", collectArgument);
+    SuggestedFix.Builder fix = replaceMethodInvocation(tree, collectMethodInvocation, state);
+    fix.merge(additionalFixes);
+
+    TypeSymbol resultTypeSymbol =
+        Optional.ofNullable(getResultType(tree)).map(Type::asElement).orElse(null);
+
+    if (Objects.isNull(resultTypeSymbol)) {
+      return SuggestedFix.emptyFix();
+    }
+
+    if (isClassValidSubstituteFor(resultTypeSymbol, Stream.class)) {
+      fix.postfixWith(tree, ".block().stream()");
+      return fix.build();
+    }
+
+    if (isClassValidSubstituteFor(resultTypeSymbol, Iterable.class)) {
+      fix.postfixWith(tree, ".block()");
+      return fix.build();
+    }
+
+    // XXX: Expected a replacement that evaluates to a type that we don't handle. Should be
+    // impossible to get here, as we only match `toStream` and `toIterable`.
+    return SuggestedFix.emptyFix();
+  }
+
+  // XXX: Assumes that the specified tree is valid, has starting position and contains the matched
+  // method invocation.
+  // XXX: Assumes that the specified tree's end is the matched method invocation's end.
+  private static SuggestedFix.Builder replaceMethodInvocation(
+      MethodInvocationTree tree, String replacement, VisitorState state) {
+    ImmutableList<ErrorProneToken> tokens =
+        ErrorProneTokens.getTokens(SourceCode.treeToString(tree, state), state.context);
+
+    int treeStartPosition = getStartPosition(tree);
+    int methodInvocationStartPosition =
+        tokens.stream()
+            .filter(token -> isTokenTheInvokedMethod(tree, token))
+            .findFirst()
+            .map(token -> treeStartPosition + token.pos())
+            .orElse(Position.NOPOS);
+    int methodInvocationEndPosition = treeStartPosition + tokens.get(tokens.size() - 1).endPos();
+
+    return SuggestedFix.builder()
+        .replace(methodInvocationStartPosition, methodInvocationEndPosition, replacement);
+  }
+
+  // XXX: Replace with prewritten solution (?) or rewrite it in a more resilient way.
+  private static boolean isTokenTheInvokedMethod(MethodInvocationTree tree, ErrorProneToken token) {
+    return token.hasName()
+        && Objects.equals(token.name().toString(), getSymbol(tree).getQualifiedName().toString());
+  }
+
+  // XXX: Replace with prewritten solution. (?)
```

Comment:
So I addressed this XXX together with the next one. Indeed there is a solution provided out-of-the-box that we can use instead of these two methods. Error Prone doesn't have that good of a documentation that easily helps us identify such improvements. We can rely on the `Types#isSubType`  here, which allows us to do pretty cool stuff. 

Will push a suggestion please take a look at it 😄. Let me know if you have any questions. I combined it with a usage of our own beautiful mini "DSL", see `MoreTypes.java`. That allows us to greatly simplify the logic used here.

This might not be easy to get at first, so if you have any questions about it, let us know :).